hera_cal/abscal.py

```diff
@@ -2436,11 +2436,15 @@ def custom_dly_slope_gain(self, gain_keys, antpos):
         antpos : type=dictionary, contains antenna position vectors. keys are ant integer, values are ant position vectors
         """
         if hasattr(self, '_dly_slope'):
-            # get dly slope dictionary
-            dly_slope = self.dly_slope[self._gain_keys[0][0]]
-            # turn delay slope into per-antenna complex gains, while iterating over gain_keys
-            return odict(list(map(lambda k: (k, np.exp(2j * np.pi * self.freqs.reshape(1, -1) * np.einsum("i...,i->...", dly_slope, antpos[k[0]][:2]))),
-                                  gain_keys)))
+            # form dict of delay slopes for each polarization in self._gain_keys
+            # b/c they are identical for all antennas of the same polarization
+            dly_slope_dict = {k[0][1]: self.dly_slope[k[0]] for k in self._gain_keys}
+
+            # turn delay slope into per-antenna complex gains, while iterating over input gain_keys
+            dly_slope_gain = odict()
+            for gk in gain_keys:
+                dly_slope_gain[gk] = np.exp(2j * np.pi * self.freqs.reshape(1, -1) * np.einsum("i...,i->...", dly_slope_dict[gk[1]], antpos[gk[0]][:2]))
```

Comment:
Can this be done with np.dot? Einsums are always kinda hard to read and usually not worth it unless there's a substantial speedup involved.

Reply:
I think einsum is the way to go here: there is a common theme across all of the functions using einsum, some of which are a little more complex (by one dimension) then this line you've picked out, so I'm gonna ask to defer that to a different issue if you feel you don't understand whats going on here

Reply:
That's fine. Maybe add a few comments to help people read off which axes are being summed over.